aqua/misc.aqua

```diff
@@ -0,0 +1,72 @@
+import "@fluencelabs/aqua-lib/builtin.aqua"
+import "registry-service.aqua"
+import "constants.aqua"
+
+alias ResourceId: string
+alias Error: string
+
+func wait(successful: *bool, len: i16, timeout: u16) -> bool:
+  status: *string
+  waiting = (arr: *bool, s: *string):
+    join arr[len - 1]
+    s <<- "ok"
+
+  waiting(successful, status)
+  par status <- Peer.timeout(timeout, "timeout")
+
+  result: *bool
+  -- HACK: workaround for #LNG-63
+  stat = status!
+  if stat == "ok":
+    result <<- true
+  else:
+    result <<- false
+
+  <- result!
+
+func appendErrors(error1: *Error, error2: *Error):
+  for e <- error2:
+    error1 <<- e
+
+func getResourceId(label: string, peer_id: string) -> ResourceId:
+  resource_id <- Registry.get_key_id(label, peer_id)
+  <- resource_id
+
+-- Get peers closest to the resource_id's hash in Kademlia network
+-- These peers are expected to store list of providers for this key
+func getNeighbours(resource_id: ResourceId) -> []PeerId:
+  k <- Op.string_to_b58(resource_id)
+  nodes <- Kademlia.neighborhood(k, nil, nil)
+  <- nodes
+
+func getResource(resource_id: ResourceId) -> ?Key, *Error:
```

Comment:
Why `getResource` returns `Key` and not `Resource`?

Reply:
Resource is a alias for the key in the terms of Resources API, but yeah better to rename getResource helper function to getKey